src/cares_wrap.cc

```diff
@@ -1935,6 +1935,27 @@ void IsIPv6(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void CanonicalizeIP(const FunctionCallbackInfo<Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+  node::Utf8Value ip(isolate, args[0]);
+  char address_buffer[sizeof(struct in6_addr)];
+  char canonical_ip[INET6_ADDRSTRLEN];
+
+  if (uv_inet_pton(AF_INET, *ip, &address_buffer) == 0) {
+    int err = uv_inet_ntop(AF_INET, address_buffer, canonical_ip,
+                           sizeof(canonical_ip));
+    CHECK_EQ(err, 0);
+
+    args.GetReturnValue().Set(String::NewFromUtf8(isolate, canonical_ip));
```

Comment:
This is a total nit:

The two if clauses seem almost identical other than AF_INET(6), could this be more like:

```cpp
  int af;
  if (uv_inet_pton(AF_INET, *ip, &address_buffer) == 0)
    af = AF_INET;
  else if (uv_inet_pton(AF_INET6, *ip, &address_buffer) == 0)
    af = AF_INET6;
  else
    return;

  int err = uv_inet_ntop(af, address_buffer, canonical_ip,
                         sizeof(canonical_ip));
  CHECK_EQ(err, 0);

  args.GetReturnValue().Set(String::NewFromUtf8(isolate, canonical_ip));
```